### DISCUSSION TITLE
LUA LCD support for Jumper T Pro

### DISCUSSION THREAD

**Discussion by tenchirocom** [EdgeTX/edgetx#3781] [LUA]
Good evening. I am writing some telemetry scripts for the Jumper T Pro 128x64 bw screen. The transmitter is at EdgeTX 2.8.4. The SD Card version is 2.8. However some of the documented lua functions are not there. Lua reports calling nil valued members for some documented functions. Some of these include lcd.drawCircle() and lcd.sizeText().

I was able to implement draw circle myself, however as I try to write my script, I keep tripping over more missing functions. Is there documentation somewhere that outlines what's not available on this transmitter? Or is there something wrong with the configuration?

Thank you.

**Comment by pfeerick** [comment]
Hm... it seems like the author who added the drawCircle function neglected to mention in the documentation comment it was colorlcd only, as is `sizeText`. Will have to update those entries 😮 

These are the B&W specific display functions:
https://github.com/EdgeTX/edgetx/blob/67e30e3893399c260f822da37aa6e4998b1aa72f/radio/src/lua/api_stdlcd.cpp#L624-L643

**Comment by tenchirocom** [comment]
Thank you so much! That is perfect. As long as I know which are there. Thanks for the lead into the source code too. I cloned the repository the other day and built it, but I hadn't started finding my way around yet. That was helpful.

Question, drawCircle is quite simple. Should I try to contribute it or is there some reason it was not included in the stdlcd api? Thank you.

**Reply by pfeerick** [reply]
I'm not sure... You're welcome to have a go at implementing and we'll find out :grin: The colorlcd version is using libopenui which is colorlcd specific, so none of those calls will be available on b&w / stdlcd, so just like with something like `lcdDrawRect` it'll have to do a bunch of fancy legwork itself.
